pkg/controller/tas/resource_flavor.go

```diff
@@ -0,0 +1,233 @@
+	for name, flavor := range h.tasCache.Clone() {
+		if isFlavorAffectedByNode(node, flavor.NodeLabels, flavor.Levels) {
+			q.AddAfter(reconcile.Request{NamespacedName: types.NamespacedName{
+				Name: string(name),
+			}}, nodeBatchPeriod)
+		}
+	}
+}
+
+func (h *nodeHandler) Generic(context.Context, event.GenericEvent, workqueue.TypedRateLimitingInterface[reconcile.Request]) {
+}
+
+func (r *rfReconciler) Reconcile(ctx context.Context, req reconcile.Request) (reconcile.Result, error) {
+	log := ctrl.LoggerFrom(ctx).WithValues("name", req.NamespacedName.Name)
+	log.V(2).Info("Reconcile TAS Resource Flavor")
+
+	flv := &kueue.ResourceFlavor{}
+	if err := r.client.Get(ctx, req.NamespacedName, flv); err != nil {
+		if client.IgnoreNotFound(err) != nil {
+			return reconcile.Result{}, err
+		}
+		r.tasCache.Delete(kueue.ResourceFlavorReference(req.NamespacedName.Name))
+	}
+	if flv.Spec.TopologyName != nil {
+		if r.tasCache.Get(kueue.ResourceFlavorReference(flv.Name)) == nil {
+			topology := kueuealpha.Topology{}
+			if err := r.client.Get(ctx, types.NamespacedName{
+				Name: *flv.Spec.TopologyName,
+			}, &topology); err != nil {
+				return reconcile.Result{}, err
+			}
+			levels := r.levels(&topology)
+			tasInfo := r.tasCache.NewFlavorCache(levels, flv.Spec.NodeLabels)
+			r.tasCache.Set(kueue.ResourceFlavorReference(flv.Name), tasInfo)
+		}
+
+		// requeue inadmissible workloads as a change to the resource flavor
+		// or the set of nodes can allow admitting a workload which was
+		// previously inadmissible.
+		if cqNames := r.cache.ActiveClusterQueues(); len(cqNames) > 0 {
+			r.queues.QueueInadmissibleWorkloads(ctx, cqNames)
+		}
+	}
+	return reconcile.Result{}, nil
+}
+
+func (r *rfReconciler) Create(event event.CreateEvent) bool {
+	rf, isRf := event.Object.(*kueue.ResourceFlavor)
+	if isRf {
+		return rf.Spec.TopologyName != nil
+	}
+	_, isNode := event.Object.(*corev1.Node)
+	return isNode
+}
+
+func (r *rfReconciler) Delete(event event.DeleteEvent) bool {
+	rf, isRf := event.Object.(*kueue.ResourceFlavor)
+	if isRf {
+		if rf.Spec.TopologyName != nil {
+			r.tasCache.Delete(kueue.ResourceFlavorReference(rf.Name))
+		}
+		return false
+	}
+	_, isNode := event.Object.(*corev1.Node)
+	return isNode
+}
+
+func (r *rfReconciler) Update(event event.UpdateEvent) bool {
+	oldRf, isOldRf := event.ObjectOld.(*kueue.ResourceFlavor)
+	newRf, isNewRf := event.ObjectNew.(*kueue.ResourceFlavor)
+	if isOldRf && isNewRf {
+		switch {
+		case ptr.Equal(oldRf.Spec.TopologyName, newRf.Spec.TopologyName):
+			return false
+		case oldRf.Spec.TopologyName == nil:
+			return true
+		default:
+			// topologyName was set so is changed or removed
+			r.tasCache.Delete(kueue.ResourceFlavorReference(*oldRf.Spec.TopologyName))
+			return newRf.Spec.TopologyName != nil
+		}
+	}
+	_, isOldNode := event.ObjectOld.(*corev1.Node)
+	_, isNewNode := event.ObjectNew.(*corev1.Node)
+	return isOldNode && isNewNode
```

Comment:
These look weird. Is there any way to enqueue the ResourceFlavor based on Node event in the eventHandler?

Reply:
Yes, we only watch RF and Nodes, we check RFs in the if, so it is enough to just respond true otherwise